### PR TITLE
Removed last Globals in logic tests and introduced architecture test

### PR DESCRIPTION
**Pull request by oscargus**

### CODE REVIEW

**Comment by Siedlerchr** [comment]
Good work! What does the architecture test enforce? That no more globals are used in logic?

File: src/test/java/net/sf/jabref/logic/importer/fetcher/IsbnFetcherTest.java [hunk 1]
```diff
@@ -3,10 +3,10 @@
 import java.io.IOException;
 import java.util.Optional;
 
-import net.sf.jabref.Globals;
 import net.sf.jabref.logic.importer.FetcherException;
 import net.sf.jabref.model.entry.BibEntry;
 import net.sf.jabref.model.entry.BibLatexEntryTypes;
+import net.sf.jabref.preferences.JabRefPreferences;
```
Comment (koppor) [inline src/test/java/net/sf/jabref/logic/importer/fetcher/IsbnFetcherTest.java]: According to https://github.com/JabRef/jabref/wiki/High-Level-Documentation#package-structure, the logic should not reference the preferences.

At least, this PR is a huge step in the direction of enforcing the idea of https://github.com/JabRef/jabref/wiki/High-Level-Documentation#package-structure.

**Comment by koppor** [comment]
Even though another nasty dependency is introduced, this PR is a huge step towards a good architecture.

**Comment by oscargus** [comment]
@Siedlerchr yes, no globals in test or model.

@koppor there are still some dependencies on JabRefPreferences in logic (not just here). Primarily in test, but also in the actual code. As I recall it, Globals.prefs is even injected in some methods, so having this dependency in test is quite OK at the moment...

**Comment by oscargus** [comment]
There are currently nine imports in `logic` (most related to preferences classes, but not all) and 42 in `test/logic` of `JabRefPreferences`.

**Comment by oscargus** [comment]
Related (to solve parts of it): #1968 #1969

**Pull request merged**